onnxruntime/core/framework/execution_frame.cc

```diff
@@ -45,6 +45,16 @@ OrtValue* IExecutionFrame::GetMutableNodeInputOrOutputMLValue(int index) {
   return const_cast<OrtValue*>(GetNodeInputOrOutputMLValue(index));
 }
 
+Status IExecutionFrame::SetOutputMLValue(int index, const OrtValue& ort_value) {
```

Comment:
Should this check that the value in the allocation plan for the index is kPreExisting? Ideally we would have something more explicit to say it's an external output that can be safely replaced though, so possibly a new enum value would be appropriate.